src/main/kotlin/de/bund/digitalservice/useid/config/SecurityConfig.kt

```diff
@@ -34,14 +36,18 @@ class SecurityConfig(
             .headers()
             .frameOptions().disable()
             .and()
-            .addFilterAfter(
-                SecurityHeadersFilter(contentSecurityPolicyProperties),
-                FilterSecurityInterceptor::class.java, // Last filter in the Spring Security filter chain
-            )
             .addFilterBefore(
-                ApiKeyAuthenticationFilter(authenticationManager),
+                TenantAuthenticationFilter(authenticationManager),
                 AnonymousAuthenticationFilter::class.java,
             )
+            .addFilterAfter(
+                ResolveTenantFilter(tenantProperties),
+                AuthorizationFilter::class.java, // Last filter in the Spring Security filter chain
+            )
+            .addFilterAfter(
+                SecurityHeadersFilter(),
+                AuthorizationFilter::class.java, // Last filter in the Spring Security filter chain
```

Comment:
To make sure the filter are added in the right order, we should use the chaining feature. So let's add the `SecurityHeaderFilter` after the `ResolveTenantFilter`.
```suggestion
                ResolveTenantFilter::class.java,
```